libp2p/messagesHandler.go

```diff
@@ -358,10 +357,11 @@ func (handler *messagesHandler) transformAndCheckMessage(pbMsg *pubsub.Message,
 	if errUnmarshal != nil {
 		// this error is so severe that will need to blacklist both the originator and the connected peer as there is
 		// no way this node can communicate with them
-		pidFrom := core.PeerID(pbMsg.From)
 		handler.blacklistPid(pid, p2p.WrongP2PMessageBlacklistDuration)
-		handler.blacklistPid(pidFrom, p2p.WrongP2PMessageBlacklistDuration)
-
+		if pbMsg != nil {
+			pidFrom := core.PeerID(pbMsg.From)
+			handler.blacklistPid(pidFrom, p2p.WrongP2PMessageBlacklistDuration)
```

Comment:
since the `ProcessReceivedMessage` is exported, I would move the check.IfNill check from the go routine as the first instruction. In this way, the code gets cleaner & testable.

Reply:
moved as discussed